### DISCUSSION TITLE
LeLink Bluetooth ODB2 adapter

### DISCUSSION THREAD

**Discussion by Frankbsmith** [collin80/SavvyCAN#468] [Q&A]
I use a LeLink Bluetooth adapter with LeafSpy which makes me believe that it is using CAN formatted data. But at $25 how is that possible since my $250 Peak adapter was the lowest priced CAN bus adapter I could find for viewing the raw CAN packets.

Someone please educate me. If the LeLink is not seeing CAN formatted data, what format does it use? Where can I find information on decoding this format?

My Peak adapter is fine for occasional diagnostic purposes. But ultimately I would love to have a small low power, Bluetooth ODB2 adapter, like my LeLink, that I could leave plugged into my car and have a Savvycan like android app that would do everything my PC and Peak adapter can do.

Thank you

**Comment by collin80** [comment]
Yes, they both connect to the CAN bus. The PeakCAN is just a lot more robust. A PeakCAN adapter can handle capturing every single frame on the bus no matter the speed. You won't be able to send a full busload of traffic through the LeLink. What they do in things like LeafSpy is key in on a few messages and filter out the rest. Bluetooth low energy is very low bandwidth - something like 9kb/sec. It's possible to add support for devices like that to SavvyCAN but the traffic you could receive would be limited by the hardware. For SavvyCAN and other such programs you are really better off with something higher end like the PeakCAN adapter. If you want something capable of high bus loads while still having the form factor of a LeLink then check out the Macchina A0: https://www.macchina.cc/catalog/a0-boards/a0-under-dash  it's currently out of stock due to the chip shortage but it's basically exactly what you are looking for. It works wirelessly with SavvyCAN.

**Comment by Frankbsmith** [comment]
Collin,

Let me start for thanking you for creating Savvycan. I love being able to
decode the CAN messages of Think City cars that I work on. I even have
people all over the world send me trace files so I can help them
troubleshoot their cars.

I do have one suggestion for making it better. A button that hides
duplicate messages. Typically when an error occurs, i need to see any other
packets where their was a transition. For example a packet may indicate
that the contactors have just opened. That is the result of the issue and
not the cause. Trying to look through the hundred plus packets between the
two that indicate when the contactors were last closed and when the opened
would be easier if I only had to look at packets that changed state.

BTW. Do you own a Think City? I assume you did since someone created the
DBC files for the Think.

Thank you and Happy New Year!

Frank Smith
410-353-8696
